client/src/Components/DashboardPage/Subscriptions/EnhancedTableRow.jsx

```diff
@@ -44,10 +46,43 @@ const Android12Switch = styled(Switch)(({ theme }) => ({
 }));
 
 export default function EnhancedTableRow({ row }) {
-  const [status, setStatus] = React.useState(row.status);
+  const [statusState, setStatusState] = useState(row.status);
+  const { enqueueSnackbar } = useSnackbar();
 
-  const handleChange = () => {
-    setStatus(!status);
+  const handleChange = async () => {
+    try {
+      setStatusState(!statusState);
```

Comment:
on handle change just `setstatus`, then in separate `useEffect` send a request whenever status change